plugins/module_utils/ilo_redfish_utils.py

```diff
@@ -231,3 +696,1561 @@ def set_wins_registration(self, mgrattr):
         if not response['ret']:
             return response
         return {'ret': True, 'changed': True, 'msg': "Modified %s" % mgrattr['mgr_attr_name']}
+
+    def validate_input_attributes(self, service_bios, attributes):
+        # Make a copy of the attributes dict
+        attrs_to_patch = dict(attributes)
+        # List to hold attributes not found
+        attrs_bad = {}
+
+        # Check the attributes
+        for attr_name, attr_value in attributes.items():
+            # Check if attribute exists
+            if attr_name not in service_bios['Attributes']:
+                # Remove and proceed to next attribute if this isn't valid
+                attrs_bad.update({attr_name: attr_value})
+                del attrs_to_patch[attr_name]
+                continue
+
+            # If already set to requested value, remove it from PATCH payload
+            if service_bios['Attributes'][attr_name] == attributes[attr_name]:
+                del attrs_to_patch[attr_name]
+
+        if attrs_bad:
+            return {
+                "ret": False,
+                "changed": False,
+                "msg": "Incorrect attributes: %s" % attrs_bad
+            }
+        # Return success with changed=False if no attrs need to be changed
+        if not attrs_to_patch:
+            return {
+                "ret": True,
+                "changed": False,
+                "msg": "Service BIOS attributes already set"
+            }
+
+        return {
+            "ret": True,
+            "msg": "Set service attributes"
+        }
+
+    def verify_drive_count(self, raid_details, logical_drives_count):
+        if len(raid_details) != logical_drives_count:
+            return {
+                "ret": False,
+                "changed": False,
+                "msg": "Logical drive count in raid details is not matching with logical drives present in the server"
+            }
+        return {
+            "ret": True,
+            "msg": "Drive count is same as input"
+        }
+
+    def set_service_bios_attributes(self, service_attributes):
```

Comment:
Getting BIOS attributes is standard Redfish functionality; we have this today as "set_bios_attributes". Why is this needed in an iLO-specific module?

Reply:
This is also a different API. Please refer the above comment.

Reply:
Removed this functionality now. Resolving this conversation, thanks!